core/src/main/java/io/questdb/cairo/FullBwdPartitionFrameCursor.java

```diff
@@ -44,37 +44,21 @@ public void calculateSize(RecordCursor.Counter counter) {
     }
 
     @Override
-    public PartitionFrame next() {
+    public PartitionFrame next(long skipTarget) {
         while (partitionIndex > -1) {
-            final long hi = reader.openPartition(partitionIndex);
+            final long hi = reader.getPartitionRowCountFromMetadata(partitionIndex);
             if (hi < 1) {
                 // this partition is missing, skip
                 partitionIndex--;
             } else {
                 frame.partitionIndex = partitionIndex;
                 frame.rowLo = 0;
                 frame.rowHi = hi;
-                partitionIndex--;
-
-                final byte format = reader.getPartitionFormat(frame.partitionIndex);
-                if (format == PartitionFormat.PARQUET) {
-                    final long addr = reader.getParquetAddr(frame.partitionIndex);
-                    assert addr != 0;
-                    final long parquetSize = reader.getParquetFileSize(frame.partitionIndex);
-                    assert parquetSize > 0;
-                    if (parquetDecoder == null) {
-                        parquetDecoder = new PartitionDecoder();
-                    }
-                    parquetDecoder.of(addr, parquetSize, MemoryTag.NATIVE_PARQUET_PARTITION_DECODER);
-                    frame.format = PartitionFormat.PARQUET;
-                    frame.parquetDecoder = parquetDecoder;
+                if (hi <= skipTarget) {
+                    partitionIndex--;
                     return frame;
```

Comment:
Don't we need to open the partition before returning the frame here and in other cursors inherited from `AbstractFullPartitionFrameCursor` and `AbstractIntervalPartitionFrameCursor`?

Also, we don't seem to reset `format` and `parquetDecoder` fields before returning the `frame`.

Reply:
we don't, may be i should document the API better? the convention is when frame is under the `skipTarget` what's required to return is the size of the frame, it would just be subtracted from the total by the main skipping logic

Reply:
Yes, it would be good to document. Also, it's a good idea to reset the fields and expose an `isInitialized()`/`isPrepared()` method, so that the caller code can check it, e.g. do an assertion.